searx/weather.py

```diff
@@ -0,0 +1,610 @@
+    if fname is None:
+        return None
+
+    ctx = "weather_symbol_url"
+    cache = get_WEATHER_DATA_CACHE()
+    origin_url = f"{YR_WEATHER_SYMBOL_URL}/{fname}.svg"
+
+    data_url = cache.get(origin_url, ctx=ctx)
+    if data_url is not None:
+        return data_url
+
+    response = network.get(origin_url, timeout=3)
+    if response and response.status_code == 200:
+        mimetype = response.headers['Content-Type']
+        data_url = f"data:{mimetype};base64,{str(base64.b64encode(response.content), 'utf-8')}"
+        cache.set(key=origin_url, value=data_url, expire=None, ctx=ctx)
+    return data_url
+
+
+@dataclasses.dataclass
+class GeoLocation:
+    """Minimal implementation of Geocoding."""
+
+    # The type definition was based on the properties from the geocoding API of
+    # open-meteo.
+    #
+    # - https://open-meteo.com/en/docs/geocoding-api
+    # - https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+    # - https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+
+    name: str
+    latitude: float  # Geographical WGS84 coordinates of this location
+    longitude: float
+    elevation: float  # Elevation above mean sea level of this location
+    country_code: str  # 2-Character ISO-3166-1 alpha2 country code. E.g. DE for Germany
+    timezone: str  # Time zone using time zone database definitions
+
+    def __str__(self):
+        return self.name
+
+    def locale(self) -> babel.Locale:
+
+        # by region of the search language
+        sxng_tag = _get_sxng_locale_tag()
+        if "-" in sxng_tag:
+            locale = babel.Locale.parse(_get_sxng_locale_tag(), sep='-')
+            print(f"FIXME: locale from search language {locale}")
+            return locale
+
+        # by most popular language in the region (country code)
+        for lang in babel.languages.get_official_languages(self.country_code):
+            try:
+                locale = babel.Locale.parse(f"{lang}_{self.country_code}")
+                print(f"FIXME: locale from geo-location {locale}")
+                return locale
+            except babel.UnknownLocaleError:
+                continue
+
+        # No locale could be determined.  This does not actually occur, but if
+        # it does, the English language is used by default.  But not region US.
+        # US has some units that are only used in US but not in the rest of the
+        # world (e.g. °F instead of °C)
+        return babel.Locale("en", territory="DE")
+
+    @classmethod
+    def by_query(cls, search_term: str) -> GeoLocation:
+        """Factory method to get a GeoLocation object by a search term.  If no
+        location can be determined for the search term, a :py:obj:`ValueError`
+        is thrown.
+        """
+
+        ctx = "weather_geolocation_by_query"
+        cache = get_WEATHER_DATA_CACHE()
+        geo_props = cache.get(search_term, ctx=ctx)
+
+        if not geo_props:
+            geo_props = cls._query_open_meteo(search_term=search_term)
```

Comment:
I'm not too sure if we really want to rely on OpenMeteo here, but I don't have a better idea either.